WGLMakie/test/message_counting.jl

```diff
@@ -0,0 +1,102 @@
+# These tests are relevant for general Makie performance but only tested here
+# because Bonito makes it easy to test. If the tests fail it is likely not
+# WGLMakies fault.
+# A lower number of messages maybe caused by optimizations or broken interactions.
+# A higher number could come from added functionality or performance regressions.
+
+
+@testset "TextBox with Menu" begin
+    f = Figure()
+    t1 = Textbox(f[1, 1])
+    m = Menu(f[2, 1], options = string.(1:1000))
+    display(edisplay, App(() -> f))
+
+    # trigger select
+    all_messages, summary_str = Bonito.collect_messages() do
+        events(f).mouseposition[] = (300, 250)
+        events(f).mousebutton[] = Makie.MouseButtonEvent(Mouse.left, Mouse.press)
+        events(f).mousebutton[] = Makie.MouseButtonEvent(Mouse.left, Mouse.release)
+    end
+    @test length(all_messages) == 44
+
+    # type text
+    for (char, expected) in zip(collect("test"), [18, 39, 39, 39])
+        _key = eval(:(Makie.Keyboard.$(Symbol(char))))
```

Comment:
Could this be getproperty instead of eval?

Reply:
yes